aten/src/ATen/native/cuda/ReduceOpsKernel.cu

```diff
@@ -222,6 +224,227 @@ void argmin_kernel_cuda(TensorIterator& iter) {
+    num_irows *= self.size(dim_);
+  }
+
+  dim3 threads(std::min(512, num_irows));
+  int maxGridDim = 1024;
+  dim3 grid(std::min(maxGridDim, num_orows), std::min(maxGridDim, cuda::ATenCeilDiv(num_irows, int(threads.x))));
+
+  tensor_kernel_scan_outer_dim_with_indices<scalar_t><<<grid, threads, 0, at::cuda::getCurrentCUDAStream()>>>(
+    self.data_ptr<scalar_t>(), values.data_ptr<scalar_t>(), indices.data_ptr<int64_t>(),
+    num_orows, num_irows, row_size, init, binary_op);
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+template <typename scalar_t, class BinaryFunction>
+__host__ void scan_innermost_dim_with_indices(const Tensor& self, Tensor& values, Tensor& indices, scalar_t init, BinaryFunction binary_op) {
+  int ndim = self.dim();
+  // Treat all outer dimensions as a single dimension.
+  int row_size = self.size(ndim - 1);
+  int num_rows = self.numel() / row_size;
+
+  dim3 threads(16, 32);
+  dim3 grid(std::min(1024, cuda::ATenCeilDiv(num_rows, int(threads.y))));
+
+  tensor_kernel_scan_innermost_dim_with_indices<scalar_t, 16, 32><<<grid, threads, 0, at::cuda::getCurrentCUDAStream()>>>(
+    self.data_ptr<scalar_t>(), values.data_ptr<scalar_t>(), indices.data_ptr<int64_t>(),
+    num_rows, row_size, init, binary_op);
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+template<typename scalar_t, typename BinaryFunction>
+void scan_dim_with_indices(const Tensor& self, Tensor& values, Tensor& indices, //int64_t dim) {
+     int64_t dim, scalar_t init, BinaryFunction binary_op) {
+  int ndim = self.dim();
+  Tensor self_ = self.contiguous();
+  Tensor values_ = values.contiguous();
+  Tensor indices_ = indices.contiguous();
+   if (dim == ndim - 1) {
+     scan_innermost_dim_with_indices<scalar_t>(self, values, indices, init, binary_op);
+   } else {
+     scan_outer_dim_with_indices<scalar_t>(self, values, indices, dim, init, binary_op);
+   }
+}
+
+void cummax_helper_cuda(const Tensor& self, Tensor& values, Tensor& indices, int64_t dim) {
+  TensorArg output_arg{ values, "output", 1 };
+  TensorArg indices_arg{ indices, "indices", 2 };
+  TensorArg input_arg{ self, "input", 3 };
+  checkAllSameGPU("cummax", {output_arg, indices_arg, input_arg});
+  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Bool, at::ScalarType::Half,
+    self.scalar_type(), "cummax_cuda", [&]() {
+    scan_dim_with_indices<scalar_t>(self, values, indices, dim, std::numeric_limits<scalar_t>::lowest(), MaxOp<scalar_t>());
+  });
+}
+
+void cummin_helper_cuda(const Tensor& self, Tensor& values, Tensor& indices, int64_t dim) {
+  TensorArg output_arg{ values, "output", 1 };
+  TensorArg indices_arg{ indices, "indices", 2 };
+  TensorArg input_arg{ self, "input", 3 };
+  checkAllSameGPU("cummin", {output_arg, indices_arg, input_arg});
+  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Bool, at::ScalarType::Half,
+    self.scalar_type(), "cummin_cuda", [&]() {
+    scan_dim_with_indices<scalar_t>(self, values, indices, dim, std::numeric_limits<scalar_t>::max(), MinOp<scalar_t>());
```

Comment:
init value is not right here, the `inf` values in the tensor will be erased by it. If you are implementing cummin/cummax for both integer and floating type, you'll need to handle them differently here.